llvm/lib/Transforms/IPO/ArgumentPromotion.cpp

```diff
@@ -640,16 +640,18 @@ static bool findArgParts(Argument *Arg, const DataLayout &DL, AAResults &AAR,
     }
 
     auto *CB = dyn_cast<CallBase>(V);
-    Value *PtrArg = cast<Value>(U);
-    if (CB && PtrArg && CB->getCalledFunction() == CB->getFunction()) {
+    Value *PtrArg = U->get();
+    if (CB && CB->getCalledFunction() == CB->getFunction() &&
+        CB->getCalledFunction()->getReturnType() ==
+            CB->getFunction()->getReturnType()) {
```

Comment:
Why is the function return type important? getCalledFunction is supposed to return null if the function types mismatch already?

Reply:
I am confused, wasn't I meant to add the check to see if the function call type doesn't match the function def?

Reply:
Ah sorry, I forgot that getCalledFunction already checks the function type. Ignore my comment about that.

Reply:
okay, cool! so I can remove that check ? and then can I get a LGTM.